Sources/StreamChat/StateLayer/ChatClient+Factory.swift

```diff
@@ -4,7 +4,20 @@
 
 import Foundation
 
-// MARK: - Factory Methods for Creating Chat Lists
+// MARK: - Factory Methods for Currently Logged In User
+
+@available(iOS 13.0, *)
+extension ChatClient {
+    /// Creates an instance of ``ConnectedUser`` which represents the logged-in user state and its actions.
+    ///
+    /// - Throws: An error if no user is currently logged-in.
+    func makeConnectedUser() async throws -> ConnectedUser {
+        let user = try await databaseContainer.backgroundRead { try CurrentUserDTO.load(context: $0) }
+        return ConnectedUser(user: user, client: self)
+    }
```

Comment:
Could we also maybe return a "ConnectedUser" once the connectUser call is finished? 🤔

Reply:
Btw, instead of exposing the error from `try CurrentUserDTO.load(context: $0)`, it is probably better to expose a clearer error that the user is not connected, and so there is not current user available 🤔

Reply:
Returning `ConnectedUser` from the `connectUser` call makes a lot of sense. The existing connect user method needs to stay the same, therefore, I could add a new async one which does that. I'll try it out but maybe a separate PR for that.

Load error thingy makes sense, I'll change that.

Reply:
I ended up adding it to this PR. I left the `makeConnectedUser` available as well although I have mixed feelings about it. Maybe it would be better to hide it and to require users to get it through the connect call. But then, it gives extra flexibility for users.
There is also a question about logout handling but I want to think about that in a separate PR. Added a note to my TODO list.